pkg/graveler/graveler.go

```diff
@@ -2526,6 +2550,31 @@ func (g *Graveler) Merge(ctx context.Context, repository *RepositoryRecord, dest
 	return commitID, nil
 }
 
+func (g *Graveler) retryRepoMetadataUpdate(ctx context.Context, repository *RepositoryRecord, f RepoMetadataUpdateFunc) error {
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = RepoMetadataUpdateMaxInterval
```

Comment:
I prefer to use an exponential backoff with a large jitter (random offset) for an operation like this.  Also note that setting MaxElapsedTime instead would control how long it takes to fail; otherwise it's really unclear!

**Rationale:** Changing repo metadata is typically not a recurring operation.  So if you and I are in a race with someone else, and always sleep for the same time, we will be in a race the next time.  Being exponential means that the average load is constant even if I try at recurring intervals; having jitter greatly increases the chance that we won't collide the next time.

Reply:
Thanks!  Actually it turns out I needed coffee ☕ and ExponentialBackoff includes a randomization factor by default!?!???!  But explicit is better than implicit.